packages/zoe/src/state.js

```diff
@@ -75,50 +76,59 @@ const makeInstallationTable = () => {
 
 // Instance Table
 // Columns: handle | installationHandle | publicAPI | terms | issuers
+// | roles | roleNames
 const makeInstanceTable = () => {
   // TODO: make sure this validate function protects against malicious
   // misshapen objects rather than just a general check.
   const validateSomewhat = makeValidateProperties(
-    harden(['installationHandle', 'publicAPI', 'terms', 'issuers']),
+    harden([
+      'installationHandle',
+      'publicAPI',
+      'terms',
+      'issuers',
+      'roles',
+      'roleNames',
+    ]),
   );
+
   return makeTable(validateSomewhat);
 };
 
 // Offer Table
-// Columns: handle | instanceHandle | issuers | payoutRules | exitRule
+// Columns: handle | instanceHandle | issuers | offerRules
 // | amounts
 const makeOfferTable = () => {
-  const insistValidPayoutRuleKinds = payoutRules => {
-    const acceptedKinds = ['offerAtMost', 'wantAtLeast'];
-    for (const payoutRule of payoutRules) {
-      assert(
-        acceptedKinds.includes(payoutRule.kind),
-        details`${payoutRule.kind} must be one of the accepted kinds.`,
-      );
-    }
+  const insistValidKeys = offerRules => {
+    const validKeys = harden(['want', 'offer', 'exit']);
+    const keys = harden(Object.getOwnPropertyNames(offerRules));
```

Comment:
use a common function for this?

Reply:
Is there a reason to? It's already only one function which is well-known.

Reply:
I won't argue with well-known, given my experience level, but it's not short. I was thinking of an alias fo r `harden(Object.getOwnPropertyNames(foo))`, for which `getKeys()` would be a vast improvement.

In addition, `Object.getOwnPropertyNames()` has other semantics that aren't relevant to what we're going to use this for. (i.e. how to deal with non-enumerable or inherited properties, since we believe that the keys for an issuer-keys holder have already been vetted.)   

Actually as I read the differences between `Object.getOwnPropertyNames()` and `Object.keys()` it looks like we ought to be able to use the latter if offerRules is valid. Would we ever want there to be a difference between those two for this kind of behavior-less object?

Reply:
I think since this function is intending to validate before saving, it's nice to use `Object.getOwnPropertyNames()` to catch any non-enumerable properties, rather than Object.keys() which will ignore them.  But it's not a big difference.